src/validation/__tests__/UniqueInputFieldNamesRule-test.ts

```diff
@@ -53,6 +53,56 @@ describe('Validate: Unique input field names', () => {
     `);
   });
 
+  it('allow and/or with duplicate fields in array', () => {
+    expectValid(`
+      {
+        field(arg: { and: [{ f: true }, { f: true }] })
+      }
+    `);
+    expectValid(`
+      {
+        field(arg: { or: [{ f: true }, { f: {f1: true} }] })
```

Comment:
This doesn't look valid as f here is both Boolean as well as an input object

Reply:
ah yeah, let me change this